www/include/views/graphs/generateGraphs/generateMetricImage.php

```diff
@@ -65,24 +65,29 @@
     $index = $_GET['index'];
 } else {
     list($hostId, $svcId) = explode('_', $_GET['svcId']);
-    if (false === is_numeric($hostId) || false === is_numeric($svcId)) {
+    if (!is_numeric($hostId) || !is_numeric($svcId)) {
         CentreonGraph::displayError();
     }
-    $query = 'SELECT id FROM index_data
-        WHERE host_id = ' . $hostId . ' AND service_id = ' . $svcId;
-    $res = $pearDBO->query($query);
-    if (PEAR::isError($res)) {
+    $res = $pearDBO->prepare(
+        'SELECT id FROM index_data
+        WHERE host_id = :hostId AND service_id = :svcId'
+    );
+    $res->bindValue(':hostId', $hostId, \PDO::PARAM_INT);
+    $res->bindValue(':svcId', $svcId, \PDO::PARAM_INT);
+    $res->execute();
+
+    if (!$res) {
         CentreonGraph::displayError();
```

Comment:
If an error occurs during execution, an exception will be thrown and this method will never be called.
Please to encapsulate `$res->execute();` into a `try catch`.

Reply:
hummm... please check this one : https://github.com/centreon/centreon/pull/7823